articles/api-auth/blacklists-vs-grants.md

```diff
@@ -0,0 +1,51 @@
+---
+description: This document covers blacklists vs. grants when it comes handling tokens.
+---
+
+# Blacklists and Client Grants
+
+Let's say that you're using a non-interactive [client](/clients) to represent your protected resource. You have a partner that calls your API, and at the end of your existing contract, you and your partner decide not to renew your partnership. As such, you now want to remove your partner's access to your API. The issue, however, is that you've given your partner an [access token](/tokens/access-token) that lasts for a month.
+
+* What can you do in this situation?
+* How might you configure your Auth0 environment to make such situations easier to handle in the future?
+
+In this article, we'll cover two methods for revoking access to the protected resource:
+
+* [Blacklisting the access token](#blacklists)
+* [Revoking the client grant](#client-grants)
+
+We will then compare the two methods and provide our recommendations.
+
+## Blacklists
+
+Let's say that you grant access to your API to anyone in possession of the appropriate access token. One method of revoking access to a user is to blacklist their token so that it can no longer be used.
+
+::: note
+Please see the Auth0 blog for an in-depth treatment on [Blacklisting JSON Web Token API Keys](https://auth0.com/blog/blacklist-json-web-token-api-keys/).
+:::
+
+Auth0-issued tokens are [JWTs](/jwt), so the identifier for each token is the JWT ID, or `jti`. With the `jti` in hand, you can make the appropriate `POST` call to the Management API's [blacklist a token endpoint](/api/management/v2#!/Blacklists/post_tokens). You'll need to provide the JWT's `aud` and `jti` claims. 
+
+Your call might look something like this:
+
+```text
+curl -H "Authorization: Bearer {JWT_API_KEY}"
+-X POST
+-H "Content-Type: application/json"
+-d '{"aud":"u6nnAxGVjbBd8etXjj554YKGAG5HuVrp","jti":"test-token"}'
+https://login.auth0.com/api/v2/blacklists/tokens
+```
+
+## Client Grants
+
+The main issue in this scenario is the length of time for which the API access token is valid: one month. 
+
+By default, Auth0 issues access tokens that last for 24 hours. Setting the token's lifetime to 24 hours means that your partner must repeat the client credentials exchange (or whichever grant you've implemented) to obtain a new access token every 24 hours. To deny access to your partner due to the expiration of your contract, you can simply delete the client grant so that when their existing token expires, they cannot request a new one.
```

Comment:
@khorne3 Shouldn't we add some info on how to delete the client grant? Probably just link to the endpoint https://auth0.com/docs/api/management/v2#!/Client_Grants/delete_client_grants_by_id and also tell them how to get the `id`?